dyn_em/module_diffusion_em.F

```diff
@@ -8044,6 +8050,9 @@ SUBROUTINE vertical_diffusion_implicit(ru_tendf, rv_tendf, rw_tendf, rt_tendf,&
      DO i = i_start, i_end
         nlflux_rho(i,k,j) = nlflux(i,k,j) * &
                            (fnm(k)*rho (i,k,j)+fnp(k)*rho (i,k-1,j))
+        IF(config_flags%use_theta_m == 1)THEN
+             nlflux_rho(i,k,j) = nlflux_rho(i,k,j)*(1.+rvovrd*moist(i,k,j,P_QV))
+        ENDIF
```

Comment:
I am not sure about this. Check if this can be done instead with the line that sets sflux(i,j) based on hfx.

Reply:
This should be equivalent. When looking at how the non-local flux is computed, you see that all terms that contribute are proportional to sflux (via the variables amf1, amf2, bmf2, hfxpbl):
https://github.com/wrf-model/WRF/blob/6d0090fe41bc4fc5a1746a420e825039c1d544b7/dyn_em/module_diffusion_em.F#L5206-L5226
I thought it would be more intuitive to view the non-local flux as being computed for dry theta and then convert it via the same equation derived in the initial PR message:
<img src="https://latex.codecogs.com/gif.latex?\overline{w'\theta_{m}'}_\mathrm{NL}\approx(1+\alpha\overline{q})\overline{w'\theta'}_\mathrm{NL}+\alpha\overline{\theta}\,\overline{w'q'}_\mathrm{NL}=(1+\alpha\overline{q})\overline{w'\theta'}_\mathrm{NL}" />
The water vapor flux term is dropped because there is no non-local water vapor flux in this scheme.

But if you still think it would be better/more readable to convert sflux instead, I can also do that.

Reply:
I added a general comment, but I will put one here too.
The nlflux terms are proportional to the surface heat flux, so it should be possible to just modify the surface heat flux the same way you did for other options. The difference from this code is that the correcting q term should be the surface one.